lib/kafka_ex.ex

```diff
@@ -39,7 +39,7 @@ defmodule KafkaEx do
   def create_worker(name, worker_init \\ []) do
     case build_worker_options(worker_init) do
       {:ok, worker_init} ->
-        Supervisor.start_child(KafkaEx.Supervisor, [worker_init, name])
+        Supervisor.start_child(KafkaEx.Supervisor, [server, worker_init, name])
```

Comment:
I think `server` maybe should have a different name.  Even knowing ahead of time what your intention was, it threw me for a loop looking for where the var `server` is defined.  `server_from_config` or `default_server` or something?  Another approach I've used that I like is to introduce a `Config` module and move all of the `get_env` stuff to that.  That way when you see `KafkaEx.Config.default_server_impl` it's pretty obvious where that value is coming from.